### PR TITLE
fix: setup and run clifus in one step

### PR DESCRIPTION
**Pull request by Greesb**
This avoids having to share bash profiles and env between steps

### CODE REVIEW

**Review by Syffe — APPROVED** [review]

File: action.yml [hunk 1]
```diff
@@ -32,9 +32,12 @@ inputs:
...
   using: "composite"
   steps:
-    - name: Install PyYAML
+    - name: Setup python venv and dependencies
       shell: bash
-      run: pip install --user PyYAML
+      run: |
+        python3 -m venv .venv
+        source .venv/bin/activate
+        python3 -m pip install PyYAML
```
Comment (sileht) [inline action.yml:41]: You need to share the same shell session between the setup and the Clifus run to make this work

Comment (sileht) [reply action.yml:41]: otherwise VIRTUALENV_PATH will not be set in next shell session.

Comment (Greesb) [reply action.yml:41]: oh :(

Comment (Greesb) [reply action.yml:41]: If we create the virtualenv in the repository calling the clifus action does it work ? @sileht

Comment (Greesb) [reply action.yml:41]: Nevermind you meant between steps in here i guess, modified the PR accordingly

File: action.yml [hunk 2]
```diff
@@ -32,12 +32,15 @@ inputs:
...
 
     - name: Run Clifus
-      shell: bash
+      shell: bash -e -l {0}
```
Comment (leocnp) [inline action.yml]: does -l helps keeping the venv active?

Comment (Greesb) [reply action.yml]: From what i understood (from https://github.com/orgs/community/discussions/25061) yes, it sources the profile of the first step.

Comment (sileht) [reply action.yml]: Why not just merge the two steps?

Comment (Greesb) [reply action.yml]: That would be too simple and i don't like simple things

Comment (Greesb) [reply action.yml]: Also there are other steps using the output from the clifus step, so if we merge them the output parsing will break for other steps.

**Review by leocnp — APPROVED** [review]

File: action.yml (hunk 2 above)
Comment (Greesb) [reply action.yml]: We could do it in one step, but it would require even more refactoring. I'll do it if this fix fails to fix the problem.

Comment (sileht) [reply action.yml]: This will not work profile is not update by pip tooling

Comment (sileht) [reply action.yml]: You have no other choice than:
- merging steps
Or
- rerunning ˋsource venv\bin\activateˋ

Comment (Greesb) [reply action.yml]: I tried to merge steps, let me know if it seems good or not now @sileht

**Review by sileht — APPROVED** [review]

**Comment by Greesb** [comment]
@sileht need manual merge, no mergify setup here

**Comment by sileht** [comment]
@mergifyio queue

**Pull request merged**